GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1263,6 +1263,24 @@ private void FillCommitInfo(GitRevision? revision)
 
         private async Task FillGpgInfoAsync(GitRevision? revision)
         {
+            // Don't show the "GPG" tab for artificial commits
+            bool showGpgInfoTab = revision?.IsArtificial != true && AppSettings.ShowGpgInformation.Value;
```

Comment:
```suggestion
            bool showGpgInfoTab = revision?.IsArtificial is false && AppSettings.ShowGpgInformation.Value;
```

Reply:
What about `null`?

Reply:
That's what `is false` means. `null` *is* not `false`, i.e. do not show the tab if we do not have a real revision.

Reply:
I am still unconvinced. These two expressions aren't equivalent:
![image](https://github.com/gitextensions/gitextensions/assets/4403806/23d01c7e-c8da-4a93-a4ba-dc7091ac5e3f)

Reply:
I'm not sure we are discussing about the logic but more on the business which is: does it make sense to display the tab when the there is no revision selected (i.e. empty repo)? I think that's not really important....

It has been decided to show the "File tree" in this case (code I borrowed) but as @mstv says, maybe it's better to not show the GPG tab in this case. Or you prefer @RussKie?

I tend to share @mstv point of view so I will change for "is false".

Reply:
`==` and `is` are equivalent - as it should be.
If it does not work here, add parens.

![image](https://github.com/gitextensions/gitextensions/assets/36601201/98f12637-4d1c-4c09-bc62-be41119383c5)

```            StringBuilder sb = new();
            sb.AppendLine("\tnull\tfalse\ttrue");
            foreach (bool? value in new bool?[] { null, false, true })
            {
                sb.AppendLine($"{value?.ToString() ?? "null"} is\t{value is null}\t{value is false}\t{value is true}")
                  .AppendLine($"{value?.ToString() ?? "null"} ==\t{value == null}\t{value == false}\t{value == true}")
                  .AppendLine($"{value?.ToString() ?? "null"} is !\t{value is not null}\t{value is not false}\t{value is not true}")
                  .AppendLine($"{value?.ToString() ?? "null"} !=\t{value != null}\t{value != false}\t{value != true}");
            }

            MessageBox.Show(sb.ToString());
```

Reply:
I feel we have had this discussion before...

Just to clarify:

`revision?.IsArtificial is false` is equivalent to `revision is not null && revision?.IsArtificial == false`

> == and is are equivalent - as it should be.

...unless you override the operator ==...